### DISCUSSION TITLE
Is it possible to change the name of an endpoint?

### DISCUSSION THREAD

**Discussion by fracturedsoftware** [orchetect/swift-midi#168] [Q&A] [answered]
Hi,

Is it possible to change the name of a virtual endpoint after it is created? I've tried a few things but it doesn't seem possible.

Thanks!

**Comment by orchetect** [comment, marked as answer]
Not currently but it's been on the to-do list. I can add some properties pretty easily and I'll push a release soon.

**Reply by fracturedsoftware** [reply]
OK, thanks.

I realise I could change the name using something like :

MIDIObjectSetStringProperty(endpoint, kMIDIPropertyName, new name as CFString)

but it would be nice to have an ‘official’ API that correctly updates everywhere necessary, including a SwiftUI popup.

I look forward to the next release

**Reply by orchetect** [reply]
Just pushed changes to main branch.

The `endpointName` property on a managed input or output is now just called `name` and it's settable.

```swift
manager.managedInputs["tag"]?.name = "New Port Name"
// or
manager.managedOutputs["tag"]?.name = "New Port Name"
```

I'll post a new release shortly once CI passes.

**Reply by fracturedsoftware** [reply]
Thank you, that works great.
